web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/CitizenCertificateRecipientConverterImpl.java

```diff
@@ -0,0 +1,44 @@
+package se.inera.intyg.intygstjanst.web.service.impl;
+
+import org.springframework.stereotype.Service;
+import se.inera.intyg.intygstjanst.web.service.CitizenCertificateRecipientConverter;
+import se.inera.intyg.intygstjanst.web.service.bean.CertificateRecipientType;
+import se.inera.intyg.intygstjanst.web.service.dto.citizen.CitizenCertificateRecipientDTO;
+import se.inera.intyg.intygstjanst.web.service.repo.RecipientRepo;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class CitizenCertificateRecipientConverterImpl implements CitizenCertificateRecipientConverter {
+
+    private final RecipientRepo recipientRepo;
+
+    public CitizenCertificateRecipientConverterImpl(RecipientRepo recipientRepo) {
+        this.recipientRepo = recipientRepo;
+    }
+
+    @Override
+    public Optional<CitizenCertificateRecipientDTO> convert(String certificateType, LocalDateTime sent) {
+        final var recipients = recipientRepo
+                .listRecipients()
+                .stream()
+                .filter(
+                        (recipient) -> recipient.getCertificateTypes().contains(certificateType)
+                        && recipient.getRecipientType() == CertificateRecipientType.HUVUDMOTTAGARE
+                )
+                .collect(Collectors.toList());
+
+        if (recipients.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of(CitizenCertificateRecipientDTO
+                .builder()
+                .id(recipients.get(0).getId())
+                .name(recipients.get(0).getName())
+                .sent(sent == null ? null : sent.toString())
```

Comment:
See comment concerning changing to a LocalDateTime instead of String. It becomes harder to make sure the LocalDateTime is serialised in a particular format when we transform it here.